tests/TypeReconciliation/ReconcilerTest.php

```diff
@@ -184,4 +183,70 @@ public function providerTestTypeIsContainedBy(): array
             ],
         ];
     }
+
+    /**
+     * @dataProvider constantAssertions
+     */
+    public function testReconciliationOfClassConstantInAssertions(string $assertion, string $expected_type): void
+    {
+        $this->addFile(
+            'psalm-assert.php',
+            '
+            <?php
+            namespace ReconciliationTest;
+            class Foo
+            {
+                const PREFIX_BAR = \'bar\';
+                const PREFIX_BAZ = \'baz\';
+                const PREFIX_QOO = Foo::PREFIX_BAR;
+            }
+            '
+        );
+        $this->project_analyzer->getCodebase()->scanFiles();
+
+        $reconciled = \Psalm\Internal\Type\AssertionReconciler::reconcile(
```

Comment:
If you're trying to transform something like `"foo"|"bar"` into an Union, I think `Type;;parseString` would be more appropriate here

Reply:
no I dont want to parse a string. I want to parse the constants and extract their types. Not sure how parseString could be of any help here? Maybe I miss something?

Reply:
nope, my mistake, I thought you were passing $expected_type in there.